File: src/main/java/com/amritan/backend/service/UserService.java
package com.amritan.backend.service;


import com.amritan.backend.dto.CreateUserRequest;
import com.amritan.backend.dto.PageResponse;
import com.amritan.backend.dto.UpdateUserRequest;
import com.amritan.backend.dto.UserResponseDto;

public interface UserService {

	UserResponseDto createUser(CreateUserRequest request);
	
	PageResponse<UserResponseDto> getAllUsers(int pageNo, int pageSize,
								String sortBy, String sortDir);
	
	UserResponseDto getUserById(Long id);
	
	UserResponseDto updateUser(Long id, UpdateUserRequest request);
	
	void deleteUser(Long id);
	
	PageResponse<UserResponseDto> getUsersByName(String keyword, int pageNo, int pageSize );
	
	
	PageResponse<UserResponseDto> getUsersByEmail(String keyword, int pageNo, int pageSize );
}

File: frontend/src/services/userService.js
import { apiRequest } from "./api";

export const createUser = async (userData) => {
  return apiRequest("/api/users", {
    method: "POST",
    body: JSON.stringify(userData),
  });
};

export const getAllUsers = async (
  pageNo = 0,
  pageSize = 10,
  sortBy = "id",
  sortDir = "asc",
) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
    sortBy,
    sortDir,
  });

  return apiRequest(`/api/users?${params}`);
};

export const getUserById = async (id) => {
  return apiRequest(`/api/users/${id}`);
};

export const updateUser = async (id, userData) => {
  return apiRequest(`/api/users/${id}`, {
    method: "PUT",
    body: JSON.stringify(userData),
  });
};

export const deleteUser = async (id) => {
  return apiRequest(`/api/users/${id}`, {
    method: "DELETE",
  });
};

export const getUsersByName = async (name, pageNo = 0, pageSize = 10) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
  });

  return apiRequest(`/api/users/name/${encodeURIComponent(name)}?${params}`);
};

export const getUsersByEmail = async (email, pageNo = 0, pageSize = 10) => {
  const params = new URLSearchParams({
    pageNo,
    pageSize,
  });

  return apiRequest(`/api/users/email/${encodeURIComponent(email)}?${params}`);
};
